File: LegendEditor/Assets/Scripts/Game/DataBase/SQLiteDataBase/SQLiteWeaponFrame.cs
using Mono.Data.Sqlite;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class WeaponFrameData
{
	public int mID;
	public string mLabel;
	public int mDirection;
	public string mAction;
	public int mFrameCount;
	public List<int> mPosX;
	public List<int> mPosY;
}

public class SQLiteWeaponFrame : SQLiteTable
{
	string COL_ID = "WeaponID";
	string COL_LABEL = "Label";
	string COL_DIRECTION = "Direction";
	string COL_ACTION = "Action";
	string COL_FRAME_COUNT = "FrameCount";
	string COL_POSX = "PosX";
	string COL_POSY = "PosY";
	public SQLiteWeaponFrame(SQLite sqlite)
		:base("WeaponFrame", sqlite)
	{}
	public void query(int id, out List<WeaponFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, COL_ID, id, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void query(int id, int direction, out List<WeaponFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, COL_ID, id, " and ");
		appendConditionInt(ref condition, COL_DIRECTION, direction, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void query(int id, int direction, string action, out List<WeaponFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, COL_ID, id, " and ");
		appendConditionInt(ref condition, COL_DIRECTION, direction, " and ");
		appendConditionString(ref condition, COL_ACTION, action, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void updateData(WeaponFrameData data)
	{
		string updateString = "";
		appendUpdateInt(ref updateString, COL_FRAME_COUNT, data.mFrameCount);
		appendUpdateIntArray(ref updateString, COL_POSX, data.mPosX);
		appendUpdateIntArray(ref updateString, COL_POSY, data.mPosY, true);
		string condition = "";
		appendConditionInt(ref condition, COL_ID, data.mID, " and ");
		appendConditionInt(ref condition, COL_DIRECTION, data.mDirection, " and ");
		appendConditionString(ref condition, COL_ACTION, data.mAction, "");
		doUpdate(updateString, condition);
	}
	public void insert(WeaponFrameData data)
	{
		string valueString = "";
		appendValueInt(ref valueString, data.mID);
		appendValueString(ref valueString, data.mLabel);
		appendValueInt(ref valueString, data.mDirection);
		appendValueString(ref valueString, data.mAction);
		appendValueInt(ref valueString, data.mFrameCount);
		appendValueIntArray(ref valueString, data.mPosX);
		appendValueIntArray(ref valueString, data.mPosY, true);
		doInsert(valueString);
	}
//-------------------------------------------------------------------------------------------------------------------------------------
protected void parseReader(SqliteDataReader reader, out List<WeaponFrameData> dataList)
	{
		dataList = new List<WeaponFrameData>();
		while (reader.Read())
		{
			WeaponFrameData data = new WeaponFrameData();
			data.mID = StringUtility.stringToInt(reader[COL_ID].ToString());
			data.mLabel = reader[COL_LABEL].ToString();
			data.mDirection = StringUtility.stringToInt(reader[COL_DIRECTION].ToString());
			data.mAction = reader[COL_ACTION].ToString();
			data.mFrameCount = StringUtility.stringToInt(reader[COL_FRAME_COUNT].ToString());
			StringUtility.stringToIntArray(reader[COL_POSX].ToString(), ref data.mPosX);
			StringUtility.stringToIntArray(reader[COL_POSY].ToString(), ref data.mPosY);
			dataList.Add(data);
		}
		reader.Close();
	}
};

File: LegendDataEditor/LegendDataEditor/Sqlite/Table/SQLiteWeaponFrame.cpp
#include "SQLiteWeaponFrame.h"
#include "Utility.h"
#include "SQLiteDataReader.h"
#include "SQLite.h"

std::string WeaponFrameData::COL_ID = "WeaponID";
std::string WeaponFrameData::COL_LABEL = "Label";
std::string WeaponFrameData::COL_DIRECTION = "Direction";
std::string WeaponFrameData::COL_ACTION = "Action";
std::string WeaponFrameData::COL_FRAME_COUNT = "FrameCount";
std::string WeaponFrameData::COL_POSX = "PosX";
std::string WeaponFrameData::COL_POSY = "PosY";

void SQLiteWeaponFrame::query(int weaponID, txVector<WeaponFrameData>& dataList)
{
	std::string conditionString;
	appendConditionInt(conditionString, WeaponFrameData::COL_ID, weaponID, "");
	std::string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
void SQLiteWeaponFrame::query(int weaponID, int direction, txVector<WeaponFrameData>& dataList)
{
	std::string conditionString;
	appendConditionInt(conditionString, WeaponFrameData::COL_ID, weaponID, " and ");
	appendConditionInt(conditionString, WeaponFrameData::COL_DIRECTION, direction, "");
	std::string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
void SQLiteWeaponFrame::query(int weaponID, int direction, const std::string& action, txVector<WeaponFrameData>& dataList)
{
	std::string conditionString;
	appendConditionInt(conditionString, WeaponFrameData::COL_ID, weaponID, " and ");
	appendConditionInt(conditionString, WeaponFrameData::COL_DIRECTION, direction, " and ");
	appendConditionString(conditionString, WeaponFrameData::COL_ACTION, action, "");
	std::string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
bool SQLiteWeaponFrame::updateData(const WeaponFrameData& data)
{
	std::string updateString;
	appendUpdateInt(updateString, WeaponFrameData::COL_FRAME_COUNT, data.mFrameCount);
	appendUpdateIntArray(updateString, WeaponFrameData::COL_POSX, data.mPosX);
	appendUpdateIntArray(updateString, WeaponFrameData::COL_POSY, data.mPosY, true);
	std::string conditionString;
	appendConditionInt(conditionString, WeaponFrameData::COL_ID, data.mID, " and ");
	appendConditionInt(conditionString, WeaponFrameData::COL_DIRECTION, data.mDirection, " and ");
	appendConditionString(conditionString, WeaponFrameData::COL_ACTION, data.mAction, "");
	return doUpdate(updateString, conditionString);
}

bool SQLiteWeaponFrame::insert(const WeaponFrameData& data)
{
	std::string valueString;
	appendValueInt(valueString, data.mID);
	appendValueString(valueString, data.mLabel);
	appendValueInt(valueString, data.mDirection);
	appendValueString(valueString, data.mAction);
	appendValueInt(valueString, data.mFrameCount);
	appendValueIntArray(valueString, data.mPosX);
	appendValueIntArray(valueString, data.mPosY, true);
	return doInsert(valueString);
}